### PR TITLE
added gRPC services

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

**Review by kirillta — APPROVED** [review]

File: HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading.Tasks;
+using CSharpFunctionalExtensions;
+using Grpc.Core;
+using Tsutsujigasaki.GrpcContracts.Models;
+using Tsutsujigasaki.GrpcContracts.Services;
+
+namespace HappyTravel.Tsutsujigasaki.Api.Services
+{
+    public class GrpcConversionService : IConversionGrpcService
```
Comment (mingazhev) [inline HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs:9]: Need to discuss whether we need `Grpc` suffix in service names

File: HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            _conversionService = conversionService;
+        }
+    
+    
+        public async Task<SingleValueConversionResponse> ConvertSingle(SingleValueConversionRequest request)
+        {
+            var (isSuccess, _, moneyAmount, error) = await _conversionService.Convert(request.SourceCurrency, request.TargetCurrency, request.Value);
+        
+            if (isSuccess)
+                return new SingleValueConversionResponse{Result = moneyAmount};
```
Comment (mingazhev) [inline HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs:22]: Weird formatting

File: HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        
+            throw new RpcException(new Status(StatusCode.FailedPrecondition, error.Detail));
```
Comment (mingazhev) [inline HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs:24]: Status code looks invalid. Maybe `Unknown` will fit better? Here and further

File: HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        }
+
+    
+        public async Task<MultipleValuesConversionResponse> ConvertMultiple(MultipleValuesConversionRequest request)
+        {
+            var (isSuccess, _, moneyAmounts, error) = await _conversionService.Convert(request.SourceCurrency, request.TargetCurrency, request.Values);
+        
+            if (isSuccess)
```
Comment (mingazhev) [inline HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs:32]: All this code duplication may be eliminate by something like:

```csharp
private static TResult ToRpcResult<TResult>(Result<TResult, ProblemDetails> result)
        {
            if (result.IsFailure)
                throw new RpcException(new Status(StatusCode.Unknown, result.Error.Detail));

            return result.Value;
        }
```

File: HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
+﻿using System.Threading.Tasks;
+using CSharpFunctionalExtensions;
+using Grpc.Core;
+using Tsutsujigasaki.GrpcContracts.Models;
+using Tsutsujigasaki.GrpcContracts.Services;
+
+namespace HappyTravel.Tsutsujigasaki.Api.Services
```
Comment (mingazhev) [inline HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs:7]: I think it is needed to move these services to another folder. These are more like a controllers than an application services

File: HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs (hunk 2 above)
Comment (DFSko) [reply HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs]: Changed

File: HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs (hunk 3 above)
Comment (DFSko) [reply HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs]: Removed

File: HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs (hunk 5 above)
Comment (DFSko) [reply HappyTravel.Tsutsujigasaki.Api/Services/GrpcConversionService.cs]: Moved to GrpcServices folder

**Review by kirillta — APPROVED** [review]

**Pull request merged**